src/reducers/collections.js

```diff
@@ -1,4 +1,4 @@
-import { OrderedMap, fromJS } from 'immutable';
```

Comment:
Causes for change in this file:

* `action.payload` is now an immutable map instead of an object.
* Logic needed a little cleanup.